lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -96,7 +99,8 @@ def transform_block_filters(self, usage_info, block_structure):
                 selected = block_keys['selected']
 
                 # Save back any changes
-                if any(block_keys[changed] for changed in ('invalid', 'overlimit', 'added')):
+                if usage_info.user.is_authenticated() and any(block_keys[changed]
```

Comment:
For edX reviewers: Library module is not supported right now. This change is just to prevent errors in case the course has one.

Reply:
Nit: The indent here is a bit awkward. I think the recommendation is to use parens around the whole statement and pull the `any` to the second line.

More importantly: This might be another place where it makes sense to push the logic down the stack. Say a new `classmethod` on `StudentModule` that essentially does `update_or_create()` like the call below, but has a check for `is_authenticated()` -- assuming of course that it's always wrong to update state for unauthenticated users (which I think is correct?).

This is a question/suggestion, and not a requirement for merging.

Reply:
I've changed the indent. As for moving the logic down the stack, I don't really know much about which level to contain the unauthenticated user check makes sense, but wouldn't moving it down increase the function call overhead when the user is not authenticated? Since we'll now be calling the `StudentModule` classmethod which then calls the `is_authenticated`, in place of just `is_authenticated`?

Reply:
I guess I just assumed that `is_authenticated` was really cheap. Is it not?